v2/devdoc/logging.md

```diff
@@ -0,0 +1,221 @@
+# `c-logging` requirements
+
+## Basic functionality
+
+`c-logging` shall expose APIs allowing the user to log events that can be used for observing the software activity, debugging, etc.
+
+The exposed API shall allow creating contexts both dynamically and on the stack and using those contexts for logging.
+
+The API shall allow easy formatting of values to be placed in the contexts by using printf-like formatting.
+
+```
+void logger_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line_no, const char* format, ...);
+
+#define LOGGER_LOG(log_level, log_context, format, ...) \
+    logger_log(log_level, log_context, __FILE__, __FUNCTION__, __LINE__, format MU_IFCOMMALOGIC(MU_COUNT_ARG(__VA_ARGS__)) __VA_ARGS__)
+
+// macro that can be used to create a dynamically allocated context
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+
+// destroy a dynamically allocated context
+#define LOG_CONTEXT_DESTROY(log_context_handle) \
+    ...
+
+// macro that can be used to create a context on the stack
+#define LOG_CONTEXT_LOCAL_DEFINE(log_context, parent_context, ...) \
+    ...
+
+// when a user wants to specify properties for a context, they can use LOG_CONTEXT_PROPERTY
+// LOG_CONTEXT_PROPERTY(property_type, property_name, property_value)
+
+// when a user wants to specify string fields in a context using printf-style formatting, they can use LOG_CONTEXT_STRING_PROPERTY
+// LOG_CONTEXT_STRING_PROPERTY(property_name, property_conversion_specifier, property_value)
```

Comment:
>property_conversion_specifier [](http://example.com/codeflow?start=46&length=29)

this is very interesting!!!! 

can I have the property_conversion_specifier as "%s=%d" and then pass "Popescu" and "42" as "property_value"?

Or is the field property_value limted to 1 value.

In that case... can be property conversion specifier be like "Address = %s"? (note the extra "Addres = ".

Or this is just a %s or %ls? If that's the case, we should have them separated, by LOG_CONTEXT_STRING_PROPERTY  /  LOG_CONTEXT_WSTRING_PROPERTY.

Reply:
LOG_CONTEXT_STRING_PROPERTY(last name, "%s%s", "ua", "ga") has been added to the test.

Reply:
When we will have wchar_t_ptr as property type we'll also gain the LOG_CONTEXT_WSTRING_PROPERTY